client/packages/common/src/ui/components/steppers/HorizontalStepper/HorizontalStepper.tsx

```diff
@@ -0,0 +1,231 @@
+      ...colors,
+    };
+
+    return <div style={style}>{icon}</div>;
+  };
+
+const getAnimationStyles = (colour: StepperColour) => {
+  const colour1 =
+    colour === 'primary' ? 'rgb(254, 147, 24, 0.4)' : 'rgba(53, 104, 212, 0.4)';
+  const colour2 = 'rgba(53, 104, 212, 0)';
+
+  return {
+    '@-webkit-keyframes pulse': {
+      '0%': {
+        '-webkit-box-shadow': `0 0 0 0 ${colour1}`,
+      },
+      '70%': {
+        '-webkit-box-shadow': `0 0 0 10px ${colour2}`,
+      },
+      '100%': {
+        '-webkit-box-shadow': `0 0 0 0 ${colour2}`,
+      },
+    },
+    '@keyframes pulse': {
+      '0%': {
+        '-moz-box-shadow': `0 0 0 0 ${colour1}`,
+        boxShadow: `0 0 0 0 ${colour1}`,
+      },
+      '70%': {
+        '-moz-box-shadow': `0 0 0 10px ${colour2}`,
+        boxShadow: `0 0 0 10px ${colour2}`,
+      },
+      '100%': {
+        '-moz-box-shadow': `0 0 0 0 ${colour2}`,
+        boxShadow: `0 0 0 0 ${colour2}`,
+      },
+    },
+  };
+};
+
+const getLabelStyle = (
+  colour: StepperColour,
+  completed?: boolean,
+  error?: boolean
+) => {
+  switch (true) {
+    case !!error:
+      return { color: 'error.main' };
+    case !!completed:
+      return { color: `${colour}.dark` };
+    default:
+      return { color: `${colour}.light` };
+  }
+};
+
+const usePaletteColour = (colour: StepperColour): PaletteColour => {
+  const theme = useTheme();
+  switch (colour) {
+    case 'secondary':
+      return {
+        light: theme.palette.gray.pale,
+        main: theme.palette.secondary.light,
+        dark: theme.palette.secondary.dark,
+        error: theme.palette.error.main,
+      };
+    default:
+      return {
+        light: theme.palette.primary.light,
+        main: theme.palette.primary.main,
+        dark: theme.palette.primary.dark,
+        error: theme.palette.error.main,
+      };
+  }
+};
+
+/* alternativeLabel shows icons on top */
+export const HorizontalStepper: FC<StepperProps> = ({
+  colour = 'secondary',
+  steps,
+}) => {
+  const paletteColour = usePaletteColour(colour);
+  const StyledConnector = getConnector(paletteColour);
+
+  return (
+    <Box flex={1}>
+      <GlobalStyles styles={getAnimationStyles(colour)} />
+      <Stepper
+        connector={<StyledConnector />}
+        activeStep={steps.findIndex(step => step?.active)}
+        orientation="horizontal"
+        alternativeLabel
+      >
+        {steps.map(step => {
+          const { active, completed, label, optional, icon, error } = step;
+          // There is no accessability role that I can find to accurately describe
+          // a stepper, so turning to testids to mark the active/completed steps
+          // for tests
+          let testId = '';
+          if (active) testId += 'active';
+          if (completed) testId += 'completed';
+          const stepIcon = icon ? undefined : getCircle(paletteColour);
+
+          return (
+            <Step
+              data-testid={testId}
+              key={label}
```

Comment:
I think index would be better as they `key` (otherwise we have to make there are no duplicates in translations for future wizards implementation steps)

Reply:
the use of index as a key in react is discouraged ([react-lists and keys](https://reactjs.org/docs/lists-and-keys.html) and [this article](https://robinpokorny.com/blog/index-as-a-key-is-an-anti-pattern/) so I tend to avoid them.

it's low risk either way - I've gone with the expectation that you wouldn't have a stepper with two steps having the same label. that would be confusing for users and a strange use case

Reply:
oki, I thought for index was fine if array is static (I assumed it would be for steppers).
my worry was that label is translated, no guarantee that there wouldn't be duplicates. From memory key is mainly used when things are sorted, so I guess it should be fine even if there are duplicates